Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/HttpContextExtensions.cs

```diff
@@ -0,0 +1,196 @@
+                    UserAgent = userAgent
+                },
+                Time = currentTime.ToString("dd/MMM/yyyy:HH:mm:ss") + " +0000",
+                TimeEpoch = currentTime.ToUnixTimeMilliseconds(),
+                RequestId = HttpRequestUtility.GenerateRequestId(),
+                RouteKey = $"{request.Method} {apiGatewayRouteConfig.Path}",
+            },
+            Version = "2.0"
+        };
+
+        if (request.Cookies.Any())
+        {
+            httpApiV2ProxyRequest.Cookies = request.Cookies.Select(c => $"{c.Key}={c.Value}").ToArray();
+        }
+
+        if (headers.Any())
+        {
+            httpApiV2ProxyRequest.Headers = headers;
+        }
+
+        httpApiV2ProxyRequest.RawQueryString = string.Empty; // default is empty string
+        if (queryStringParameters.Any())
+        {
+            // this should be decoded value
+            httpApiV2ProxyRequest.QueryStringParameters = queryStringParameters;
+
+            // this should be the url encoded value and not include the "?"
+            // e.g. key=%2b%2b%2b
+            httpApiV2ProxyRequest.RawQueryString = HttpUtility.UrlPathEncode(request.QueryString.Value?.Substring(1));
+
+        }
+
+        if (pathParameters.Any())
+        {
+            // this should be decoded value
+            httpApiV2ProxyRequest.PathParameters = pathParameters;
+        }
+
+        if (HttpRequestUtility.IsBinaryContent(request.ContentType))
+        {
+            // we already converted it when we read the body so we dont need to re-convert it
+            httpApiV2ProxyRequest.IsBase64Encoded = true;
+        }
+
+        return httpApiV2ProxyRequest;
+    }
+
+    /// <summary>
+    /// Translates an <see cref="HttpContext"/> to an <see cref="APIGatewayProxyRequest"/>.
+    /// </summary>
+    /// <param name="context">The <see cref="HttpContext"/> to be translated.</param>
+    /// <param name="apiGatewayRouteConfig">The configuration of the API Gateway route, including the HTTP method, path, and other metadata.</param>
+    /// <returns>An <see cref="APIGatewayProxyRequest"/> object representing the translated request.</returns>
+    public static APIGatewayProxyRequest ToApiGatewayRequest(
+        this HttpContext context,
+        ApiGatewayRouteConfig apiGatewayRouteConfig)
+    {
+        var request = context.Request;
+        var body = HttpRequestUtility.ReadRequestBody(request);
+        var contentLength = HttpRequestUtility.CalculateContentLength(request, body);
+
+        var pathParameters = RouteTemplateUtility.ExtractPathParameters(apiGatewayRouteConfig.Path, request.Path);
+
+        var (headers, multiValueHeaders) = HttpRequestUtility.ExtractHeaders(request.Headers);
+        var (queryStringParameters, multiValueQueryStringParameters) = HttpRequestUtility.ExtractQueryStringParameters(request.Query);
+
+        if (!headers.ContainsKey("content-length"))
```

Comment:
Shouldn't we always update the content-length since we are updating the body?

Reply:
yea thats a good point. it doesnt hurt to always update it

Reply:
actually i just checked this. api gateway doesnt set this if the request doesnt contain it already

Reply:
looks like rest api doesnt set this automatically by default